### PR TITLE
Retry if server responds with 500 error

### PR DESCRIPTION
**Pull request by sbmaxx**
I noticed that sometimes our selenium grid can response with 500 error and webdriver.io doesn't retry. Also i added logs for that

### CODE REVIEW

**Comment by sbmaxx** [comment]
Here how it looks like in teamcity with `request.debug = true`.

We send command to SG, it response with error 500 and webdriver didn't handle it. So it handing before it has been killed due timeout.

```
[10:14:43][test name] [10:14:43]:  COMMAND	POST 	 "/wd/hub/session"
[10:14:43][test name] [10:14:43]:  DATA		 {"desiredCapabilities":{"browserName":"chrome","version":"","javascriptEnabled":true,"locationContextEnabled":true,"handlesAlerts":true,"rotatable":true,"platform":"ANY","loggingPrefs":{"browser":"ALL","driver":"ALL"},"requestOrigins":{"url":"http://webdriver.io";,"version":"3.4.0","name":"webdriverio"}}}
[10:14:43][test name] REQUEST { timeout: 10000,
[10:14:43][test name]   body: '{"desiredCapabilities":{"browserName":"chrome","version":"","javascriptEnabled":true,"locationContextEnabled":true,"handlesAlerts":true,"rotatable":true,"platform":"ANY","loggingPrefs":{"browser":"ALL","driver":"ALL"},"requestOrigins":{"url":"http://webdriver.io";,"version":"3.4.0","name":"webdriverio"}}}',
[10:14:43][test name]   headers: 
[10:14:43][test name]    { Connection: 'keep-alive',
[10:14:43][test name]      Accept: 'application/json',
[10:14:43][test name]      'User-Agent': 'webdriverio/webdriverio/3.4.0',
[10:14:43][test name]      'Content-Type': 'application/json; charset=UTF-8',
[10:14:43][test name]      'Content-Length': 305 },
[10:14:43][test name]   followAllRedirects: true,
[10:14:43][test name]   json: true,
[10:14:43][test name]   method: 'POST',
[10:14:43][test name]   uri: 
[10:14:43][test name]    { protocol: 'http:',
[10:14:43][test name]      slashes: true,
[10:14:43][test name]      auth: 'AUTH',
[10:14:43][test name]      host: 'SELENIUM_GRID:4444',
[10:14:43][test name]      port: '4444',
[10:14:43][test name]      hostname: 'SELENIUM_GRID',
[10:14:43][test name]      hash: null,
[10:14:43][test name]      search: null,
[10:14:43][test name]      query: null,
[10:14:43][test name]      pathname: '/wd/hub/session',
[10:14:43][test name]      path: '/wd/hub/session',
[10:14:43][test name]      href: 'http://GRID_URL:4444/wd/hub/session' },
[10:14:43][test name]   callback: [Function],
[10:14:43][test name]   tunnel: false }
[10:14:43][test name] REQUEST make request http://GRID_URL:4444/wd/hub/session
[10:14:43][test name] REQUEST onRequestResponse http://GRID_URL:4444/wd/hub/session/6c92c9913909c9b8b333a3843ba1eca36a4c4eb0-f2cb-4a2b-b325-77b46063381d/execute 500 { server: 'nginx',
[10:14:43][test name]   date: 'Thu, 07 Jan 2016 07:14:43 GMT',
[10:14:43][test name]   'content-type': 'application/json;charset=UTF-8',
[10:14:43][test name]   'content-length': '13458',
[10:14:43][test name]   connection: 'keep-alive',
[10:14:43][test name]   expires: 'Thu, 01 Jan 1970 00:00:00 GMT',
[10:14:43][test name]   'cache-control': 'no-cache',
[10:14:43][test name]   'set-cookie': [ 'uid=CmajrVaOEGPASkHSKTKTAg==; path=/' ] }
[10:14:43][test name] REQUEST finish init function http://GRID_URL:4444/wd/hub/session/6c92c9913909c9b8b333a3843ba1eca36a4c4eb0-f2cb-4a2b-b325-77b46063381d/execute
[10:14:43][test name] REQUEST response end http://GRID_URL:4444/wd/hub/session/6c92c9913909c9b8b333a3843ba1eca36a4c4eb0-f2cb-4a2b-b325-77b46063381d/execute 500 { server: 'nginx',
[10:14:43][test name]   date: 'Thu, 07 Jan 2016 07:14:43 GMT',
[10:14:43][test name]   'content-type': 'application/json;charset=UTF-8',
[10:14:43][test name]   'content-length': '13458',
[10:14:43][test name]   connection: 'keep-alive',
[10:14:43][test name]   expires: 'Thu, 01 Jan 1970 00:00:00 GMT',
[10:14:43][test name]   'cache-control': 'no-cache',
[10:14:43][test name]   'set-cookie': [ 'uid=CmajrVaOEGPASkHSKTKTAg==; path=/' ] }
[10:14:43][test name] REQUEST end event http://GRID_URL:4444/wd/hub/session/6c92c9913909c9b8b333a3843ba1eca36a4c4eb0-f2cb-4a2b-b325-77b46063381d/execute
[10:14:43][test name] REQUEST has body http://GRID_URL:4444/wd/hub/session/6c92c9913909c9b8b333a3843ba1eca36a4c4eb0-f2cb-4a2b-b325-77b46063381d/execute 13458
[10:14:43][test name] REQUEST emitting complete http://GRID_URL:4444/wd/hub/session/6c92c9913909c9b8b333a3843ba1eca36a4c4eb0-f2cb-4a2b-b325-77b46063381d/execute
[10:31:28][test name] The build 1. Автосборка dev и пул реквестов::3.0 E2E тесты, desktop #6ecceb7 {buildId=2330598} has been running for more than 20 minutes. Terminating...
[10:31:29][test name] Dumping threads before termination:
[10:31:29][test name] PID: 11650, PPID: 1756, Command line: /bin/bash /home/buildfarm/tmpfs/tca05.i.fog.yandex.net/temp/agentTmp/custom_script5453055782313823560
[10:31:29][test name] Failed to locate jps tool. No thread dump information is available
[10:31:29][test name] 
[10:31:29][test name] PID: 11653, PPID: 11650, Command line: node bin/fiji e2e --url=http://URL --browser=desktop-chrome --reporter=mocha-multi --remote-grid --concat --debug e2e
[10:31:29][test name] Failed to locate jps tool. No thread dump information is available
[10:31:29][test name] 
```

**Comment by christian-bromann** [comment]
@sbmaxx cool, ping me if you think this is ready for review. Thanks

**Comment by sbmaxx** [comment]
@christian-bromann i'm ready

File: lib/utils/RequestHandler.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 var Q = require('q'),
     url = require('url'),
-    request = require('request'),
+    request = require('request-promise'),
```
Comment (christian-bromann) [inline lib/utils/RequestHandler.js]: You still use `request-promise` here

**Comment by sbmaxx** [comment]
@christian-bromann fixed

**Comment by christian-bromann** [comment]
LGTM!

**Pull request merged**